include/ghex/rma/thread/access_guard.hpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2020, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+#ifndef INCLUDED_GHEX_RMA_THREAD_ACCESS_GUARD_HPP
+#define INCLUDED_GHEX_RMA_THREAD_ACCESS_GUARD_HPP
+
+#include <mutex>
+#include <condition_variable>
+#include <memory>
+#include "../access_mode.hpp"
+#include "./handle.hpp"
+
+namespace gridtools {
+namespace ghex {
+namespace rma {
+namespace thread {
+
+struct local_access_guard
```

Comment:
Comments on the use of this class and what an epoch is?

Reply:
will do